src/compressed_chunk.c

```diff
@@ -216,17 +216,36 @@ size_t Compressed_DelRange(Chunk_t *chunk, timestamp_t startTs, timestamp_t endT
     return deleted_count;
 }
 
-static Chunk *decompressChunk(CompressedChunk *compressedChunk) {
-    Sample sample;
+// decompress chunk
+static Chunk *decompressChunk(const CompressedChunk *compressedChunk) {
+	assert(compressedChunk != NULL);
+
     uint64_t numSamples = compressedChunk->count;
     Chunk *uncompressedChunk = Uncompressed_NewChunk(numSamples * SAMPLE_SIZE);
+    Sample *samples = uncompressedChunk->samples;
 
     ChunkIter_t *iter = Compressed_NewChunkIterator(compressedChunk, CHUNK_ITER_OP_NONE, NULL);
-    for (uint64_t i = 0; i < numSamples; ++i) {
-        Compressed_ChunkIteratorGetNext(iter, &sample);
-        Uncompressed_AddSample(uncompressedChunk, &sample);
-    }
+
+	// 4 samples per iteration
+	uint64_t i = 0;
+	uint64_t n = numSamples / 4;
```

Comment:
@swilly22 making this const might improve perf on some very rare compilers and also might be more clear

Reply:
const uint64_t n = numSamples / 4;